src/PrestaShopBundle/Controller/Admin/Improve/Design/ImageSettingsController.php

```diff
@@ -0,0 +1,374 @@
+        $grid = $gridFactory->getGrid($filters);
+
+        $generationSettingsForm = $this->get('prestashop.admin.image_settings.image_generation_options.form_handler')->getForm();
+
+        $thumbnailRegenerationForm = $this->createForm(ThumbnailRegenerationType::class);
+        $helper = $this->get('prestashop.adapter.image_settings.helper');
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/ImageSettings/index.html.twig', [
+            'imageTypeGrid' => $this->presentGrid($grid),
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'generationSettingsForm' => $generationSettingsForm->createView(),
+            'thumbnailRegenerationForm' => $thumbnailRegenerationForm->createView(),
+            'showMoveForm' => $helper->showMovingForm(),
+            'showDuplicatesAlert' => $helper->showDuplicatesAlert(),
+            'duplicatesFolder' => $helper->getProductImagesDir() . 'duplicates/',
+        ]);
+    }
+
+    /**
+     * Process image generation options configuration form.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update', 'create', 'delete'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_image_settings_index"
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function saveOptionsAction(Request $request): RedirectResponse
+    {
+        $generationSettingsFormHandler = $this->get('prestashop.admin.image_settings.image_generation_options.form_handler');
+
+        $generationSettingsForm = $generationSettingsFormHandler->getForm();
+        $generationSettingsForm->handleRequest($request);
+
+        if ($generationSettingsForm->isSubmitted()) {
+            $errors = $generationSettingsFormHandler->save($generationSettingsForm->getData());
+
+            if (empty($errors)) {
+                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_image_settings_index');
+            }
+
+            $this->flashErrors($errors);
+        }
+
+        return $this->redirectToRoute('admin_image_settings_index');
+    }
+
+    /**
+     * Provides filters functionality.
+     *
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request): RedirectResponse
+    {
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.core.grid.definition.factory.image_type'),
+            $request,
+            ImageTypeGridDefinitionFactory::GRID_ID,
+            'admin_image_settings_index'
+        );
+    }
```

Comment:
No need if that uses the common search

Reply:
Grid definition factory isn't using filterable parent, so common search will not work.